### PR TITLE
:sparkles: Serialize Reference Features

### PR DESCRIPTION
**Pull request by ThomasParistech**
- Make "Reference" a dataclass
- Add descriptor setter from numpy for the "Reference" class
- Add I/O utils

N.B. It could be useful to have pycolmap::Track and pycolmap::TrackElement behave like dataclasses (constructor with kwargs)

### CODE REVIEW

File: pixsfm/localization/main.py [hunk 1]
```diff
@@ -380,11 +380,12 @@ def __init__(
             self.target_reference_funcs[self.conf.target_reference]
 
         self.references = references
+        # Assure that extractor is valid
+        if self.extractor is None:
+            self.extractor = FeatureExtractor(self.conf.dense_features)
```
Comment (ThomasParistech) [inline pixsfm/localization/main.py:385]: Required to extract the query fmaps

**Comment by Phil26AT** [comment]
Hi @ThomasParistech, thanks for the PR! It looks great. 

Re pycolmap: Yes good idea, pycolmap's Track is still a bit clumsy to use.

**Comment by ThomasParistech** [comment]
> Hi @ThomasParistech, thanks for the PR! It looks great.
> 
> Re pycolmap: Yes good idea, pycolmap's Track is still a bit clumsy to use.

Let me know if there remain some things to fix in this PR :)

**Comment by Phil26AT** [comment]
HI @ThomasParistech, yes we should merge this now. One thing: Could you add `store_references` to [\_\_init__.py](https://github.com/ThomasParistech/pixel-perfect-sfm/blob/serialize-reference-features/pixsfm/features/__init__.py)? I guess this is ready then. Thanks a lot for your work!

**Pull request merged**